src/Functions/dateDiff.cpp

```diff
@@ -167,9 +173,13 @@ class DateDiffImpl
     template <typename TransformX, typename TransformY, typename T1, typename T2>
     Int64 calculate(const TransformX & transform_x, const TransformY & transform_y, T1 x, T2 y, const DateLUTImpl & timezone_x, const DateLUTImpl & timezone_y) const
     {
+        UInt8 day_of_week_mode = enable_default_monday_first ? 0 : 3;
```

Comment:
Let's not invent yet another name for the same thing, let's use `UInt8 week_mode = ...`.